dashboard/dashapi/dashapi.go

```diff
@@ -402,11 +403,21 @@ type BugReport struct {
 type Asset struct {
 	Title       string
 	DownloadURL string
-	// We unfortunately cannot make it be of asset.Type, we get a circular dependency
-	// in that case.
-	Type string
+	Type        AssetType
 }
 
+type AssetType string
+
+// Asset types used throughout the system.
+// DO NOT change them, this will break compatibility with DB content.
+const (
+	BootableDisk       AssetType = "bootable_disk"
```

Comment:
This is of type string only for syz-ci json config, right?

If so, I would switch this to integer codes and add `asset.TypeDescription.configName` for the config.
Then per-type configs can be retrieved with `c.Assets[assetTypes[assetType].configName]`.

Reply:
This also makes it easier to e.g. debug-print data structures and to manually explore the DB/raw dashboard requests. Are there substantial reasons to prefer integers in this case?

Reply:
Strings are for humans. Numbers are for machines.
These are stored both in DB and in the config. This means we can never ever rename config names. And people don't always read comments that something shouldn't be renamed.

If you want debug printing add `String() string` method. fmt/log will use it.
And we will be able to reuse this method to access the config as `c.Assets[assetType.String()]`.

Reply:
> And people don't always read comments that something shouldn't be renamed.

One more risk I see is that we become dependent on the exact order of asset type consts (in case of `iota`) or take extra care when assigning int values ourselves. So it's tricky either way.

If the biggest problem is in renaming config-file-facing consts, maybe we could still save strings in the DB, and (once it's needed) just add something like `customConfigName` to `TypeDescription` in `pkg/asset/type.go` and adjust `pkg/asset/config.go` to be able to understand it?

Reply:
// During offline discussion it was decided to leave this as is.